services/common/rs_server_common/data_retrieval/eodag_provider.py

```diff
@@ -0,0 +1,80 @@
+"""EODAG Provider."""
+from pathlib import Path
+
+from eodag import EODataAccessGateway, EOProduct
+
+from .provider import CreateProviderFailed, Product, Provider, TimeRange
+
+
+class EodagProvider(Provider):
+    """An EODAG provider.
+
+    It uses EODAG to provide data from external sources.
+    """
+
+    def __init__(self, config_file: str, provider: str):
+        """Create a EODAG provider.
+
+        :param config: the eodag configuration
+        """
+        self.provider: str = provider
+        self.client: EODataAccessGateway = self.init_eodag_client(config_file)
+        self.client.set_preferred_provider(self.provider)
+
+    def init_eodag_client(self, config_file: Path):
+        """Initialize the eodag client.
+
+        The EODAG client is initialized for the given provider.
+
+        :param config_file: the path to the eodag configuration file
+        :return: the initialized eodag client
+        """
+        try:
+            return EODataAccessGateway(config_file.as_posix())
+        except Exception as e:
+            raise CreateProviderFailed(f"Can't initialize {self.provider} provider") from e
+
+    def _specific_search(self, between: TimeRange) -> dict[str, Product]:
+        """TODO Check/Update "between" time format."""
+        products, _ = self.client.search(start=str(between.start), end=str(between.end), provider=self.provider)
```

Comment:
you forget the parameter "raise_errors=True"
that forces Eodag to raise the exceptions
that are raised during the search.